### ISSUE TITLE
Systray corrupted after awesome.restart()

### ISSUE THREAD

**Issue by spheenik** [awesomeWM/awesome#1738]
Output of `awesome --version`:

```
➜  ~ awesome --version     
awesome v4.1 (Technologic)
 • Compiled against Lua 5.3.4 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.5
 • LGI version: 0.9.1
```

**How to reproduce the issue:**

Start awesome normally. Icons in the systray are shown all right.
Press Window-Ctrl-R, to restart awesome.

**Actual result:**

After restart, the systray icons (background) are screwed up.
[Screenshots](https://cloud.schrodt.org/index.php/s/wUiRBn1kVxEiNi4)
(I can produce all kinds of patterns, looks like it re-renders them on an uncleared buffer)

**Expected result:**

Systray icons are nice and clean :)

Background is set to black, no transparency.
I tried everything in inssue #359, to no avail. I also read the documentation on the systray widget, but nothing helped.

I hope you have a remedy. Thank you!

**Comment by spheenik** [comment]
Hmm, I found #1720 after I posted this, and setting a forced_width for the systray did indeed help.

I would have no problem giving it a width, if it would draw the icons aligned to the right. Is there a way to accomplish this?

Feel free to close as a duplicate.

**Comment by spheenik** [comment]
@psychon: You mentioned (in #359 iirc) that you think this behaviour is a bug in QT, which was fixed in 5.5 or 5.6. I run Arch Linux, which looks like it runs 5.8 already. The icons are from Nextcloud and Enpass, both of which are using QT 5.

**Comment by Elv13** [comment]
Duplicate of #1720
